### PR TITLE
JAMES-2393 Introduce dedicated modules for event sourcing

### PR DESCRIPTION
**Pull request by mbaechler**
This intend to ease future usage, external usage as well as allow a clear view.

### CODE REVIEW

**Comment by mbaechler** [comment]
test this please

**Comment by chibenwa** [comment]
```
[a779469aa938b525463937e74313a0f7e63c2ea2] Tests in error: 
[a779469aa938b525463937e74313a0f7e63c2ea2]   NetworkMatcherIntegrationTest.remoteAddrInNetworkShouldSupportRangesDefinedByAMiddleIp:165 » ConditionTimeout
```

test this please

File: event-sourcing/event-store-api/src/test/java/org/apache/james/eventsourcing/eventstore/EventStoreTest.java [hunk 1]
```diff
@@ -1,28 +1,22 @@
-package org.apache.james.eventsourcing;
+package org.apache.james.eventsourcing.eventstore;
```
Comment (chibenwa) [inline event-sourcing/event-store-api/src/test/java/org/apache/james/eventsourcing/eventstore/EventStoreTest.java]: Missing license spotted...

**Comment by chibenwa** [comment]
```
[a779469aa938b525463937e74313a0f7e63c2ea2] Tests in error: 
[a779469aa938b525463937e74313a0f7e63c2ea2]   TestDiscardProtocol.testComplete:94 »  Timed out wait for be notified that rea...
```

test this please

**Comment by chibenwa** [comment]
```
[a779469aa938b525463937e74313a0f7e63c2ea2] Tests in error: 
[a779469aa938b525463937e74313a0f7e63c2ea2]   ToRepositoryTest.userShouldBeAbleToAccessReprocessedMails:130 » ConditionTimeout
```

test this please

**Comment by chibenwa** [comment]
test this please

File: event-sourcing/event-sourcing-core/pom.xml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    </parent>
+
+    <artifactId>event-sourcing-core</artifactId>
+
+    <name>Apache James :: Event sourcing :: core</name>
+    <description>James Event Sourcing system</description>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.james</groupId>
```
Comment (rouazana) [inline event-sourcing/event-sourcing-core/pom.xml]: ${project.groupId} :) (x3)

File: event-sourcing/event-store-api/src/test/java/org/apache/james/eventsourcing/eventstore/EventStoreTest.java (hunk 1 above)
Comment (rouazana) [reply event-sourcing/event-store-api/src/test/java/org/apache/james/eventsourcing/eventstore/EventStoreTest.java]: still missing?

File: mailbox/plugin/quota-mailing-cassandra/pom.xml [hunk 3]
```diff
@@ -34,6 +34,27 @@
     <description>Apache James Mailbox Cassandra implementation of Quota mailing listener</description>
 
     <dependencies>
+        <dependency>
+            <groupId>org.apache.james</groupId>
```
Comment (rouazana) [inline mailbox/plugin/quota-mailing-cassandra/pom.xml]: {project.groupId} x3

File: mailbox/plugin/quota-mailing-memory/pom.xml [hunk 4]
```diff
@@ -34,6 +34,27 @@
     <description>Apache James Mailbox memory implementation of Quota mailing listener</description>
 
     <dependencies>
+        <dependency>
+            <groupId>org.apache.james</groupId>
```
Comment (rouazana) [inline mailbox/plugin/quota-mailing-memory/pom.xml]: idem

File: mailbox/plugin/quota-mailing/pom.xml [hunk 5]
```diff
@@ -34,6 +34,20 @@
     <description>Apache James Mailbox Quota mailing listener</description>
 
     <dependencies>
+        <dependency>
+            <groupId>org.apache.james</groupId>
```
Comment (rouazana) [inline mailbox/plugin/quota-mailing/pom.xml]: idem

File: event-sourcing/event-store-cassandra/pom.xml [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+        </dependency>
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>event-sourcing-event-store-api</artifactId>
+            <type>test-jar</type>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>${project.groupId}</groupId>
+            <artifactId>event-sourcing-core</artifactId>
```
[the remark this answers is not in the document]
Comment (aduprat) [reply event-sourcing/event-store-cassandra/pom.xml]: Same on other _pom.xml_ files containing the _event-sourcing-core_ artifact.

**Comment by aduprat** [comment]
```
Tests in error: 
  SpamAssassinContract.deletingSpamMessageShouldNotImpactSpamAssassinState:538 » ConditionTimeout
  SpamAssassinContract.spamShouldBeDeliveredInSpamMailboxOrInboxWhenMultipleRecipientsConfigurations:608 » ConditionTimeout
```
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

**Comment by mbaechler** [comment]
merged

**Pull request closed without merging**